src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py

```diff
@@ -456,26 +470,30 @@ def setData(self, data_item, is_batch=False):
         self.cmdCalculateBg.setEnabled(True)
         self.cmdExtrapolate.setEnabled(True)
 
-        self.model.setItem(W.W_GUINIERA, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_GUINIERB, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_PORODK, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_PORODSIGMA, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_CORETHICK, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_INTTHICK, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_HARDBLOCK, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_CRYSTAL, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_POLY, QtGui.QStandardItem(""))
-        self.model.setItem(W.W_PERIOD, QtGui.QStandardItem(""))
-
-        self._canvas.data = data
-        self._canvas.extrap = None
-        self._canvas.draw_q_space()
-        self.cmdTransform.setEnabled(False)
-        self._path = data.name
-        self._realplot.data = None
-        self._realplot.draw_real_space()
-
-        self.model.setItem(W.W_FILENAME, QtGui.QStandardItem(self._path))
+        try:
+            self.model.itemChanged.disconnect(self.model_changed)
+            self.model.setItem(W.W_GUINIERA, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_GUINIERB, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_PORODK, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_PORODSIGMA, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_CORETHICK, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_INTTHICK, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_HARDBLOCK, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_CRYSTAL, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_POLY, QtGui.QStandardItem(""))
+            self.model.setItem(W.W_PERIOD, QtGui.QStandardItem(""))
+
+            self._canvas.data = data
+            self._canvas.extrap = None
+            self._canvas.draw_q_space()
+            self.cmdTransform.setEnabled(False)
+            self._path = data.name
+            self._realplot.data = None
+            self._realplot.draw_real_space()
+
+            self.model.setItem(W.W_FILENAME, QtGui.QStandardItem(self._path))
+        finally:
+            self.model.itemChanged.connect(self.model_changed)
```

Comment:
If you move the update to `W.W_FILENAME` with the rest of the updates, then you don't need the try-finally block.

Rather than calling `draw_q_space()` directly, you could instead call the suppressed `model_changed()` as is done elsewhere.